app/controllers/concerns/course/lesson_plan/personalization_concern.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+module Course::LessonPlan::PersonalizationConcern
+  extend ActiveSupport::Concern
+
+  # Dispatches the call to the correct personalization algorithm
+  # If the algorithm takes too long (e.g. voodoo AI magic), it is responsible for scheduling an async job
+  def update_personalized_timeline_for(course_user, timeline_algorithm = nil)
+    timeline_algorithm ||= course_user.timeline_algorithm
+    # Rails genius: match enum with string...
+    case timeline_algorithm
```

Comment:
You can use `send` to call a method name from a string instead of using a `case` statement.
https://stackoverflow.com/questions/8036446/ruby-convert-string-to-method-name